Resolving 'Make the admin page for games more robust #135' via Games/admin page

Abebes01 opened the pull request:
Resolving #135 by allowing partial searching and added Meta class to some objects for plural conjugations.

Abebes01 committed:
Adding new metrics for sorting

Abebes01 committed:
Making the admin page for games more robust

MuyanXie commented:
@ekabaeva we need your review before we can proceed

ekabaeva reviewed (approved):
The list of sorting metrics is sufficient, and it works correctly. Looks good to me!

ekabaeva reviewed (changes requested):
The original implementation works perfectly. I would just print the string that is used for the search somewhere after the search is performed because it might be good for a user to see the string they used the most recently (it can be either in a search bar or next to the Games Title). It might be also useful to add a button somewhere that can redirect the user to the original games grid because if they want to sort the whole game grid after they perform the search, they have to go all the way back to the original grid because sorting allows sorting among the search results.

The task itself asks for the feature to be implemented for the admin. Currently, the feature is available for all registered users and I think we should leave it as it is, but we might need to reconsider the task.

pauline196 committed:
Merge branch 'dev' into games/admin_page

Abebes01 committed:
resolving merge conflicts

Abebes01 committed:
Merge branch 'dev' into games/admin_page

Abebes01 committed:
Add search query display and View All Games button

As requested by reviewer, implemented a display of search query text next to the Games title for better user experience. Also added a View All Games button to allow users to easily return to the full games grid after performing a search.

Abebes01 commented:
Hello @ekabaeva I have added the requested additions. Let me know if there are any other issues.

ekabaeva reviewed (approved):
I tested the implementation, and all the comments were addressed and implemented. Great job!

MuyanXie committed:
Merge branch 'dev' into games/admin_page

MuyanXie commented:
@Abebes01 I not able to verify the changes. When I load the fixtures games-fixture-500.json and the test_players.json, I can login to the games page. However, when selecting sort by rating lowest first, I encountered this error:

FieldError at /games/
Cannot resolve keyword 'review' into field. Choices are: BGG_id, achievement, avg_rating, categories, complexity, description, expected_playtime, game_lists, guide, id, image, interactivefictiongame, leaderboardprivacysetting, leaderboards, lobby, match, matchproposal, max_players, max_playtime, mechanics, metrics, min_players, min_playtime, name, people, popularity, published_guide_id, published_guide_id_id, publishers, rating_percentage, reviews, rules, suggested_age, tournament, twine_file, users, year_published

MuyanXie reviewed (changes requested):
see above error messages

ekabaeva committed:
Merge branch 'dev' into games/admin_page

ekabaeva committed:
Merge branch 'dev' into games/admin_page
